### DISCUSSION TITLE
Query regarding API Rules

### DISCUSSION THREAD

**Discussion by pranjaldugar** [pocketbase/pocketbase#2415] [Q&A] [answered]
Hi,

I have 3 collections,
parents (auth),
students (auth),
addresses (base);

the student collection contails a relation field for address, and a relation field for parents (upto 2 can be selected),

In my app, I want to allow **parent** to view only those **address** that the students have in relation.
basically I am ...("students").getFullList({expand: "address, parents"});  [when parent is loged in]
___

in API Rule of **addresses**,
`@collection.students.parents.id ?= @request.auth.id` 
--> this is giveing ability to view all address irrespective of the students having that address or not;
I want parents to be able to view only those where the students have in relation.
what should be done?

Thanks a ton!

**Comment by ganigeorgiev** [comment, marked as answer]
If the relation between student-address is in the `students` collection then you'll have to use the `@collection.*` filter. If it is in the `addresses` collection then you can use dot-notation (_dot-notation for indirect relations will be also supported in the future_).

As far as I understand from your description, you seems to have the first case. An example `addresses` rule could be:

```js
@collection.students.address ?= id &&
@collection.students.parents.id ?= @request.auth.id
```

**Reply by pranjaldugar** [reply]
oh,
I did try the below code but it was not working
```
@collection.students.address.id = id &&
@collection.students.parents.id ?= @request.auth.id
```

Thanks for the help!
